### PR TITLE
Fix re-applying v1beta1 TaskRun failing 🚢

### LINKED ISSUE #2276: applying the same TaskRun yaml more than once produces errors
I am using the following TaskRun yaml
```
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: echo-hello-world-task-run
spec:
  taskSpec:
    steps:
      - name: echo
        image: ubuntu
        command:
          - echo
        args:
          - "Hello World"

```
When I do a `kubectl apply -f <file>` on this TaskRun more than once I get errors

# Expected Behavior
Upon applying the file, if I made changes to the TaskRun, apply the changes. Otherwise kubernetes should tell me that it's unchanged, like this:
```
taskrun/echo-hello-world-task-run unchanged
```

# Actual Behavior
Applying the file when nothing exists works, but any additional applies causes errors, regardless of whether or not I change the contents of the file. I noticed that if I change the name of the TaskRun it works (a new TaskRun with the new name is created).

# Steps to Reproduce the Problem
I confirm I have no TaskRuns
```
zjgoodma@Big Business ~/t/tutorial> kubectl get tr
No resources found in default namespace.
```
I create the TaskRun and confirm it ran successfully.
```
zjgoodma@Big Business ~/t/tutorial> kubectl apply -f hello-world-taskrun.yaml
taskrun.tekton.dev/echo-hello-world-task-run created

zjgoodma@Big Business ~/t/tutorial> kubectl get tr
NAME                        SUCCEEDED   REASON      STARTTIME   COMPLETIONTIME
echo-hello-world-task-run   True        Succeeded   12s         7s
```
I apply the exact same TaskRun yaml again without making any changes to it and get an error
```
zjgoodma@Big Business ~/t/tutorial> kubectl apply -f hello-world-taskrun.yaml
Error from server (BadRequest): error when applying patch:
{"spec":{"taskSpec":{"steps":[{"args":["Hello World"],"command":["echo"],"image":"ubuntu","name":"echo"}]}}}
to:
Resource: "tekton.dev/v1beta1, Resource=taskruns", GroupVersionKind: "tekton.dev/v1beta1, Kind=TaskRun"
Name: "echo-hello-world-task-run", Namespace: "default"
Object: &{map["apiVersion":"tekton.dev/v1beta1" "kind":"TaskRun" "metadata":map["annotations":map["kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"tekton.dev/v1beta1\",\"kind\":\"TaskRun\",\"metadata\":{\"annotations\":{},\"name\":\"echo-hello-world-task-run\",\"namespace\":\"default\"},\"spec\":{\"taskSpec\":{\"steps\":[{\"args\":[\"Hello World\"],\"command\":[\"echo\"],\"image\":\"ubuntu\",\"name\":\"echo\"}]}}}\n" "pipeline.tekton.dev/release":"devel"] "creationTimestamp":"2020-03-24T21:03:24Z" "generation":'\x02' "labels":map["app.kubernetes.io/managed-by":"tekton-pipelines"] "name":"echo-hello-world-task-run" "namespace":"default" "resourceVersion":"14257" "selfLink":"/apis/tekton.dev/v1beta1/namespaces/default/taskruns/echo-hello-world-task-run" "uid":"63bcc1fa-22c9-4534-b0c7-4a477eba0b84"] "spec":map["inputs":map[] "outputs":map[] "serviceAccountName":"" "taskSpec":map["steps":[map["args":["Hello World"] "command":["echo"] "image":"ubuntu" "name":"echo" "resources":map[]]]] "timeout":"1h0m0s"] "status":map["completionTime":"2020-03-24T21:03:29Z" "conditions":[map["lastTransitionTime":"2020-03-24T21:03:29Z" "message":"All Steps have completed executing" "reason":"Succeeded" "status":"True" "type":"Succeeded"]] "podName":"echo-hello-world-task-run-pod-2l9xg" "startTime":"2020-03-24T21:03:24Z" "steps":[map["container":"step-echo" "imageID":"docker-pullable://ubuntu@sha256:bec5a2727be7fff3d308193cfde3491f8fba1a2ba392b7546b43a051853a341d" "name":"echo" "terminated":map["containerID":"docker://3ced6cc7b6988dd2a1c88c563a5b4279497e2aaaab7ea5b6774cbc696e0670d6" "exitCode":'\x00' "finishedAt":"2020-03-24T21:03:28Z" "reason":"Completed" "startedAt":"2020-03-24T21:03:28Z"]]]]]}
for: "hello-world-taskrun.yaml": admission webhook "webhook.pipeline.tekton.dev" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "inputs"
```

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

```
zjgoodma@Big Business ~/t/tutorial> kubectl version
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-13T11:51:44Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```

- Tekton Pipeline version:

  **Output of `tkn version`:**

```
zjgoodma@Big Business ~/t/tutorial> tkn version
Client version: 0.8.0
Pipeline version: v0.11.0-rc2
```

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

Re-apply a v1beta1 resource would failed because of `inputs` (or
`outputs`) not a valid field. This is mainly because we are using
`v1alpha1` as stored version and thus getting the struct as v1alpha1
internally.

This fixes this issue by marking `TaskRun.Spec.Inputs` and
`TaskRun.Spec.Outputs` as pointers. It doesn't change the API, but
fixes the problem as a `nil` pointer will not be unmarshall where the
struct was.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

Fixes #2276 

/cc @sbwsg @bobcatfish @afrittoli 

/priority critical-urgent
/kind bug

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Fix re-applying v1beta1 TaskRun failing 🚢
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,6 @@ go 1.13
 
 require (
 	cloud.google.com/go v0.47.0 // indirect
-	cloud.google.com/go/storage v1.0.0
```
Comment (afrittoli) [inline go.mod]: I'm not sure why this change?

File: go.mod [hunk 2]
```diff
@@ -43,7 +42,6 @@ require (
 	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
 	golang.org/x/tools v0.0.0-20200214144324-88be01311a71 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.1.0 // indirect
-	google.golang.org/api v0.15.0
```
Comment (afrittoli) [inline go.mod]: Ditto

File: go.mod (hunk 1 above)
Comment (vdemeester) [reply go.mod]: Me neither.. `go mod` wonders… it comes, it goes those two things.. I have no idea why…

**Comment by vdemeester** [comment]
> Thank you for this!
> Were you able to prove that this fixes the issue?
> /lgtm

Yeah, re-applying twice the same `TaskRun` yaml failed before, succeeds now. I am not entirely sure how to write a tests for this as it is because of a `kubectl` behaviour (well I know how we could do, but it means adding another form of tests that we don't have yet).
